contracts/src/NStates.sol

```diff
@@ -58,6 +58,17 @@ contract NStates is IncrementalMerkleTree {
         insertLeaf(h);
     }
 
+    /*
+     * Game deployer has the ability to initialize players onto the board.
+     */
+    function spawn(uint256 h, uint256 n) public onlyOwner {
```

Comment:
should call set()

nullifier should also be called `rho` for consistency. just realized the NewNullifier event calls it `nf`, whoops. Could you change that to `rho` as well